filament/src/materials/dof/dof.mat

```diff
@@ -181,11 +200,10 @@ void initBucket(out Bucket ring) {
 
 void initRing(const float i, const float ringCount, const float kernelSize, const vec2 noise,
         out float offset, out float count, out mat2 r, out vec2 p) {
-    const float ringDensity = float(RING_DENSITY);
     float radius = (kernelSize / (ringCount - 0.5)) * i;
 
     // this is never called with i == 0
-    count = ringDensity * i;
+    count = 8.0 * i;
```

Comment:
Why 8?

Reply:
Oh, this is because we want to create a (2k+1)*(2k+1) box, that is a n*n grid with n odd. Each "ring" is the outside layer of that box. So ring 0 obviously is the 1x1 box, ring 1 is the 3x3 box, which has 8 samples on its outside layer. This is where the 8 comes from, then all subsequent rings, are built from ring 1.
Each "box" is then "morphed" to the shape we want for the bokeh (i.e. a circle).

Reply:
That would be a perfect comment for that line :p